host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncModelOperationHandlerWrapper.java

```diff
@@ -127,6 +128,12 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
         // Validate references on the new model
         final Resource domain = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS);
+
+        // for HCs running in admin-only=true, we don't try to update master
+        if (context.getRunningMode() == RunningMode.ADMIN_ONLY) {
+            return;
+        }
```

Comment:
Thanks, updated in new commit: https://github.com/luck3y/wildfly-core/commit/275a91bcef68fa685867b414422c7de4b35d2ab7